uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/calendar-ctrl.js

```diff
@@ -0,0 +1,283 @@
+      this._fetchMonthsFromPeriod();
+      this.refresh();
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {object} absenceType
+     * @return {object} style
+     */
+    this.getAbsenceTypeStyle = function (absenceType) {
+      return {
+        backgroundColor: absenceType.color,
+        borderColor: absenceType.color
+      };
+    };
+
+    /**
+     * Returns day name of the sent date(Monday, Tuesday etc.)
+     *
+     * @param  {string} date
+     * @return {string}
+     */
+    this.getDayName = function (date) {
+      var day = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+      return day[this._getDateObjectWithFormat(date).day()];
+    };
+
+    /**
+     * Decides whether sent date is a public holiday
+     *
+     * @param  {string} date
+     * @return {boolean}
+     */
+    this.isPublicHoliday = function (date) {
+      return !!this._publicHolidays[this._getDateObjectWithFormat(date).valueOf()];
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    this.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Converts given date to moment object with server format
+     *
+     * @param {Date/String} date from server
+     * @return {Date} Moment date
+     */
+    this._getDateObjectWithFormat = function (date) {
+      return moment(date, sharedSettings.serverDateFormat).clone();
+    };
+
+    /**
+     * Find the month which matches with the sent date
+     * and return the related object
+     *
+     * @param {object} date
+     * @param {array} months
+     * @return {object}
+     */
+    this._getMonthObjectByDate = function (date, months) {
+      return _.find(months, function (month) {
+        return (month.month === date.month()) && (month.year === date.year());
+      });
+    };
+
+    /**
+     * Returns the styles for a specific leaveRequest
+     * which will be used in the view for each date
+     *
+     * @param  {object} leaveRequest
+     * @param  {object} dateObj - Date UI object which handles look of a calendar cell
+     * @return {object}
+     */
+    this._getStyles = function (leaveRequest, dateObj) {
+      var absenceType,
+        status = this._leaveRequestStatuses[leaveRequest.status_id];
+
+      if (status.name === 'waiting_approval'
+        || status.name === 'approved'
+        || status.name === 'admin_approved') {
+        absenceType = _.find(this.absenceTypes, function (absenceType) {
+          return absenceType.id == leaveRequest.type_id;
+        });
+
+        //If Balance change is positive, mark as Accrued TOIL
+        if (leaveRequest.balance_change > 0) {
+          dateObj.UI.isAccruedTOIL = true;
+          return {
+            border: '1px solid ' + absenceType.color
```

Comment:
Why is it necessary to specify the `1px solid`? That should in the css, and only the `border-color` property should be specified